static/js/formula.js

```diff
@@ -260,25 +258,58 @@ function is_valid_formula(str) {
 
         //If the total length of segments obtained from the string are not equal to the original
         //string, then there is an invalid piece.
-        isValid = (acceptedSegmentLengths == (str.length-form_qty.length));
+        if(/^\d+/g.test(str)){
+            isValid = (acceptedSegmentLengths == (str.length-form_qty.length));
+        }else{
+            isValid = acceptedSegmentLengths == str.length;
+        }
```

Comment:
Checks string to decide how to check the total segment length vs str length. Even if the formula begins with something like 1, this should validate it appropriately.